geomstats/visualization.py

```diff
@@ -549,3 +566,89 @@ def plot(points, ax=None, space=None,
         klein_disk.draw(ax, **point_draw_kwargs)
 
     return ax
+
+
+class DataArrow():
+    """Arrow class for pointing from some data coordinates to others."""
+
+    def __init__(self, fig):
+        self.fig = fig
+
+    def draw(self, x_from, y_from, x_to, y_to):
+        """
+        Draw the arrow in data coordinates.
+
+        :param x_from: x coordinate at arrow base
+        :param y_from: y coordinate at arrow base
+        :param x_to: x coordinate at arrow head
+        :param y_to: y coordinate at arrow head
+        """
+        fig_trans = self.fig.transFigure.inverted()
+        coord_from = fig_trans.transform(
+            self.fig.axes[0].transData.transform(
+                (x_from, y_from)))
+        coord_to = fig_trans.transform(
+            self.fig.axes[0].transData.transform(
+                (x_to, y_to)))
+        arrow = matplotlib.patches.FancyArrowPatch(
+            coord_from,
+            coord_to,
+            transform=self.fig.transFigure,
+            fc="k",
+            connectionstyle="arc3,rad=0.",
+            arrowstyle='fancy',
+            alpha=1.,
+            mutation_scale=5.)
+        self.fig.patches.append(arrow)
+
+
+class Ellipsis2D():
+    """Plotting class for ellipses representing SPD matrices in 2D."""
+
+    def __init__(self):
+        self.fig = plt.figure()
+        spec = gridspec.GridSpec(ncols=1, nrows=1, figure=self.fig)
+        self.fig.add_subplot(spec[0, 0])
+        self.colors = ['r', 'g', 'b', 'xkcd:camel', 'k']
+        self.colors_alt = ['xkcd:burgundy', 'olive', 'cyan', 'xkcd:mud brown']
+        nice_matplotlib()
+
+    def draw(self, data_point, **kwargs):
+        """Draw the ellipsis."""
+        X, Y, x, y = self.compute_coordinates(data_point)
+        self.fig.axes[0].plot(X, Y, **kwargs)
+        return x, y
+
+    @staticmethod
+    def compute_coordinates(point):
+        """
+        Compute the ellipsis coordinates from 2D SPD matrix.
+
+        :param point: array-like, shape = [2, 2]: SPD matrix.
+        :return:
+        X: array-like, shape = [Np,];
+        Y: array-like, shape = [Np,];
+        X and Y coordinates.
+        """
+        w, vr = gs.linalg.eig(point)
+        w = w.real + EPS
+        Np = 100
+
+        [e1, e2] = w
+        x0, y0 = 0, 0
+        n = vr.shape[0]
+        angle = SpecialOrthogonal(n).angle_of_rot2(vr)
+        c, s = gs.cos(angle), gs.sin(angle)
+        the = gs.linspace(0, 2 * gs.pi, Np)
+        X = e1 * gs.cos(the) * c - s * e2 * gs.sin(the) + x0
+        y = e1 * gs.cos(the) * s + c * e2 * gs.sin(the) + y0
+        return X, y, X[Np // 4], y[Np // 4]
+
+    def plot(self):
+        """Plot final plot."""
```

Comment:
This title should maybe go in the `examples/plot_mdm...` file? I might want to use your Ellipsis class to plot something that is not related to `mdm` in a near future.

Reply:
Agreed, I moved the title setting in the example file.